### PR TITLE
Created the events cog for event handling

### PR DESCRIPTION
**Pull request by zaidi-01**
Created an event cog for event handling with 'on_message' event and updated aryas.py for startup extensions

### CODE REVIEW

File: src/extensions/events.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    Checks for Discord server links in a message
+    then kicks members if links are found
+    """
+    if message.author != self.bot.user:
+      if 'Moderator' not in message.author.roles:
+        pattern = 'discord.gg/[a-z0-9A-Z]+'
+         if re.search(pattern, message.content)
+            await self.bot.delete_message(message)
+            await self.bot.send_message(message.author, 'Advertising is not allowed without permission!')
+            await self.bot.kick(message.author)
```
Comment (tom-sherman) [inline src/extensions/events.py]: Put this code in a shared method that is called here and in the `?kick` command, or invoke the `kick` method directly.

Use the private message text as the reason for the kick.

File: src/extensions/events.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import discord
+from discord.ext import commands
+import re
+
+class Events:
+  def __init__(self, bot):
+    self.bot: commands.Bot = Bot
+  
+  @commands.event
+  async def on_message(self, message):
```
Comment (tom-sherman) [inline src/extensions/events.py]: At present, this coroutine is not very extendable. Let's have a discussion on the best way to make this more maintainable

**Review by tom-sherman — CHANGES_REQUESTED** [review]

File: src/utility.py [hunk 3]
```diff
@@ -1,10 +1,19 @@
 import asyncio
-
+import discord
 from discord.ext import commands
-
 from src.globals import MESSAGE_SLEEP_TIME
 
 
+async def kick_user(user, mod, server, bot, reason):
```
Comment (tom-sherman) [inline src/utility.py]: Add docstrings for this function

File: src/utility.py [hunk 4]
```diff
@@ -1,10 +1,19 @@
...
+    channel = discord.utils.get(server.channels, name='mod_log')
+    try:
+        await bot.ban(message.author)
```
Comment (tom-sherman) [inline src/utility.py]: This needs to be kick, not ban

File: src/utility.py [hunk 5]
```diff
@@ -1,10 +1,19 @@
...
+        msg = '{} was kicked by {}. Reason: {}'.format(user, mod, reason)
+        await bot.send_message(channel, msg)
+    except Exception as e:
+        print(e)
+        await bot.send_message(channel, 'Failed to kick {} for {}'.format(user, reason))
```
Comment (tom-sherman) [inline src/utility.py]: Use the `send` function in this file.

File: src/utility.py [hunk 6]
```diff
@@ -1,10 +1,19 @@
...
 
 
+async def kick_user(user, mod, server, bot, reason):
+    channel = discord.utils.get(server.channels, name='mod_log')
+    try:
+        await bot.ban(message.author)
+        msg = '{} was kicked by {}. Reason: {}'.format(user, mod, reason)
+        await bot.send_message(channel, msg)
+    except Exception as e:
+        print(e)
```
Comment (tom-sherman) [inline src/utility.py]: Use the logger, don't print to stdout

File: src/utility.py [hunk 7]
```diff
@@ -1,10 +1,19 @@
 import asyncio
-
+import discord
 from discord.ext import commands
-
 from src.globals import MESSAGE_SLEEP_TIME
 
 
+async def kick_user(user, mod, server, bot, reason):
+    channel = discord.utils.get(server.channels, name='mod_log')
```
Comment (tom-sherman) [inline src/utility.py]: Remove this line for now, address it in subsequent pull request

File: src/extensions/events.py [hunk 8]
```diff
@@ -15,10 +16,13 @@ def __init__(self, bot):
     if message.author != self.bot.user:
       if 'Moderator' not in message.author.roles:
         pattern = 'discord.gg/[a-z0-9A-Z]+'
-         if re.search(pattern, message.content)
+         if re.search(pattern, message.content):
```
Comment (tom-sherman) [inline src/extensions/events.py]: Add comments to explain the process of what the bot is doing. First he deletes, then messages, then kicks etc.

File: src/extensions/events.py [hunk 9]
```diff
@@ -15,10 +16,13 @@ def __init__(self, bot):
...
+            reason = 'Advertising without permission'
             await self.bot.delete_message(message)
             await self.bot.send_message(message.author, 'Advertising is not allowed without permission!')
-            await self.bot.kick(message.author)
+            await kick_user(message.author.name, self.bot.user.mention, message.server, self.bot, reason)
+    
+    await self.bot.process_commands(message)
```
Comment (tom-sherman) [inline src/extensions/events.py]: Add a comment explaining why we need this.

**Review by tom-sherman — CHANGES_REQUESTED** [review]

File: src/utility.py (hunk 3 above)
Comment (zaidi-01) [reply src/utility.py]: Done

File: src/utility.py (hunk 4 above)
Comment (zaidi-01) [reply src/utility.py]: Fixed

File: src/utility.py (hunk 5 above)
Comment (zaidi-01) [reply src/utility.py]: Done

**Review by tom-sherman — APPROVED** [review]

**Comment by tom-sherman** [comment]
This has now been merged.

**Pull request closed without merging**